File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    
    const sum = number.reduce((acc, item) => acc + item);
    const len  =discount.length;
    
    for(let key = 0; key <= len - sum; key++) {
        let arr = number.map((item, idx) => item);
        let flag = true;
        
        for(let i = key; i<sum + key; i++) {
            const item = discount[i];
            const idx = want.indexOf(item);
            
            if(idx === -1) {
                flag = false;
                break;
            }
            
            if(arr[idx] <= 0) {
                flag = false;
                break;
            } else {
                arr[idx] -=1;
            }
            
            
        }
        
        if(flag) answer +=1;
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    int sum = 0;
    int wantLen = want.size();
    int len = discount.size();
    
    for(auto& item : number)
    {
        sum += item;
    }
    
    for(int first = 0; first <= len - sum ; first++) {
        vector<int> vec(wantLen);
        bool flag = true;
        
        for(int i=first; i < first + sum; i++){
            auto loc = find(want.begin(), want.end(), discount[i]);
            
            if(loc == want.end()) {
                flag = false;
                break;
            }
            
            int idx = loc- want.begin();
            
            if(vec[idx] == number[idx]) {
                flag = false;
                break;
            } else vec[idx] +=1;
        }
        
        if(flag) answer +=1;
    }
    
    
    
    return answer;
}